apps/gdaltindex_lib.cpp

```diff
@@ -1167,218 +1366,123 @@ GDALTileIndexOptionsNew(char **papszArgv,
+            psOptions->ymin = (*oTe)[1];
+            psOptions->xmax = (*oTe)[2];
+            psOptions->ymax = (*oTe)[3];
         }
-        else if (EQUAL(papszArgv[iArg], "-fetch_md"))
+
+        if (auto fetchMd =
+                argParser->present<std::vector<std::string>>("-fetch_md"))
         {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(3);
-            GDALTileIndexRasterMetadata md;
-            md.osRasterItemName = papszArgv[++iArg];
-            md.osFieldName = papszArgv[++iArg];
-            const char *pszType = papszArgv[++iArg];
-            if (EQUAL(pszType, "String"))
-                md.eType = OFTString;
-            else if (EQUAL(pszType, "Integer"))
-                md.eType = OFTInteger;
-            else if (EQUAL(pszType, "Integer64"))
-                md.eType = OFTInteger64;
-            else if (EQUAL(pszType, "Real"))
-                md.eType = OFTReal;
-            else if (EQUAL(pszType, "Date"))
-                md.eType = OFTDate;
-            else if (EQUAL(pszType, "DateTime"))
-                md.eType = OFTDateTime;
-            else
+            if (fetchMd->size() % 3)
             {
-                CPLError(CE_Failure, CPLE_NotSupported, "Unknown type '%s'",
-                         pszType);
+                CPLError(CE_Failure, CPLE_AppDefined,
+                         "-fetch_md requires at least 3 arguments and up to "
+                         "3*n: %d were given.",
+                         static_cast<int>(fetchMd->size()));
                 return nullptr;
             }
-            psOptions->aoFetchMD.emplace_back(std::move(md));
-        }
-        else if (papszArgv[iArg][0] == '-')
-        {
-            CPLError(CE_Failure, CPLE_NotSupported, "Unknown option name '%s'",
-                     papszArgv[iArg]);
-            return nullptr;
-        }
-        else
-        {
-            if (psOptionsForBinary)
+
+            // Loop
+            for (size_t i = 0; i < fetchMd->size(); i += 3)
             {
-                if (!psOptionsForBinary->bDestSpecified)
+                OGRFieldType type;
+                const auto typeName{fetchMd->at(i + 2)};
+                if (typeName == "String")
+                {
+                    type = OFTString;
+                }
+                else if (typeName == "Integer")
                 {
-                    psOptionsForBinary->bDestSpecified = true;
-                    psOptionsForBinary->osDest = papszArgv[iArg];
+                    type = OFTInteger;
+                }
+                else if (typeName == "Integer64")
+                {
+                    type = OFTInteger64;
+                }
+                else if (typeName == "Real")
+                {
+                    type = OFTReal;
+                }
+                else if (typeName == "Date")
+                {
+                    type = OFTDate;
+                }
+                // "Time" is not documented: is it really unsupported?
```

Comment:
in practice for an image catalog, a time without date is of little use

Reply:
removed